### ISSUE TITLE
W3C Markup validation error on layout attribute.

### ISSUE THREAD

**Issue by gladsonrobinson** [angular/material#3453]
Am looking for my app to be w3c standard validation error free. 
Currently am getting an error as follow when I validate my app at https://validator.w3.org/nu/

######  Error: Attribute layout-align not allowed on element div at this point.
ect...

Please let me know is there any possible way to overcome this.

**Comment by danwkennedy** [comment]
You can prefix custom attributes with "data-" and they should pass. So
"layout" would be "data-layout". This is due to Angular adding custom HTML
attributes that a W3C inspector wouldn't recognize.

**Comment by gustavohenke** [comment]
> So "layout" would be "data-layout".

Nope, `layout` is an CSS-only attribute.

Anyway, these W3C Validators aren't "valid" anymore. They won't tell you anything useful, and they probably won't work with Web Components (which are very like Angular directives, so...)

**Comment by gladsonrobinson** [comment]
Have tried data-layout but its not working as expected. 
Is there any workaround that can be done for this particular problem as its my requirement to keep the w3c standard.

**Comment by DavidSouther** [comment]
**tl;dr** you cannot use Angular Material and the w3c validator together.

The Angular 1.x framework and related projects are pushing the boundaries of the web forward. This includes the ability to create and define new HTML entities (tags) and attributes at runtime - which is something that cannot be validated. (The tool has no way to check what tags and attributes are legal without running all of Angular and your code.) This is also true with the new Web Components standard. Until such a time as w3c's specs and tooling captures the goals of Angular, Web Components, and related frameworks, the fundamental incompatibilities are unable to be overcome.

_PS._ Most custom attributes you can get around with `data-` prefix, but `layout` depends on CSS behaviors and will not work; nor will custom tags (directives).

**Comment by Splaktar** [comment]
For anyone still looking into this issue, the `layout` CSS has been changed to use classes instead of attributes. This means that in many cases, you can just use the classes (i.e. `class="layout-row"` or `class="flex-66"`) instead of the attributes.

This likely won't get your validator completely passing, but it might be a step in the right direction if that's something you are investigating.
